.github/PULL_REQUEST_TEMPLATE/pull_request_template.md

```diff
@@ -0,0 +1,20 @@
+# Pull Request Checklist
+
+Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:
+
+See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).
+
+- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
+- [ ] Doxygen docs
+- [ ] Unit tests
```

Comment:
```suggestion
- [ ] Unit tests
- [ ] Ran tests locally
- [ ] Portable (works in Win, Lin Mac)
```

Reply:
>  Ran tests locally

I don't think that is a necessary requirement. Some contributors rely on CI to do the testing.
If you are referring to general - make sure your changes build and  your tests run, I think that is a given (so not needed in the checklist). I have never found myself forgetting those fundamental steps :p

> Portable (works in Win, Lin Mac)

I usually don't build/test locally for all the various platforms and versions that we support and I consider that the job of the CI. That is too high a bar. Has this been a recurring issue for you, where you need to be reminded the code needs to be portable up front? Having CI fail and it letting the author know what went wrong seems reasonable.

Most changes we make are platform/OS agnostic anyway, right?